test/utils/pickers-utils.tsx

```diff
@@ -74,6 +80,123 @@ export function createPickerRenderer({
   };
 }
 
-export function openMobilePicker() {
-  fireEvent.click(screen.getByRole('textbox'));
-}
+type OpenPickerParams =
+  | {
+      type: 'date' | 'date-time' | 'time';
+      variant: 'mobile' | 'desktop';
+    }
+  | {
+      type: 'date-range';
+      variant: 'mobile' | 'desktop';
+      initialFocus: 'start' | 'end';
+    };
+
+export const openPicker = (params: OpenPickerParams) => {
+  switch (params.variant) {
+    case 'desktop': {
+      switch (params.type) {
+        case 'date':
+        case 'date-time': {
+          return userEvent.mousePress(screen.getByLabelText(/choose date/i));
+        }
+
+        case 'time': {
+          return userEvent.mousePress(screen.getByLabelText(/choose time/i));
+        }
+
+        case 'date-range': {
+          return fireEvent.focus(
+            screen.getAllByRole('textbox')[params.initialFocus === 'start' ? 0 : 1],
+          );
+        }
+
+        default: {
+          throw Error('Invalid params');
+        }
+      }
+    }
+
+    case 'mobile': {
+      switch (params.type) {
+        case 'date':
+        case 'date-time':
+        case 'time': {
+          return userEvent.mousePress(screen.getByRole('textbox'));
+        }
+
+        case 'date-range': {
+          return userEvent.mousePress(
```

Comment:
`userEvent.focus` does not work, I don't know why
I don't want to change this behavior now since we have issues showing that the whole "open on focus" / "close on blur" has problems that will need deeper changes soon.

Reply:
I might be explained by https://github.com/mui/mui-x/blob/ebb0e3b1a4d940b02ca20abd61ade86fce9b6e94/packages/x-date-pickers-pro/src/DateRangePicker/DateRangePickerInput.tsx#L154-L155

`openOnFocus` is only true on desktop.

Reply:
Maybe
This PR is big enough as it is but we will investigate